### PR TITLE
Add lint 'discarded_futures'.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/test/debugger/program_explorer_test.dart [hunk 1]
```diff
@@ -78,6 +78,8 @@ void main() {
       when(fakeServiceManager.connectedApp!.isDartWebAppNow).thenReturn(false);
 
       final mockScriptManager = MockScriptManager();
+      //`then` is used
```
Comment (CoderDake) [inline packages/devtools_app/test/debugger/program_explorer_test.dart:81]: you could probably just make setUpAll's closure async:
`    setUpAll(() async {`

File: packages/devtools_app/test/inspector/inspector_error_navigator_test.dart [hunk 2]
```diff
@@ -68,6 +68,8 @@ void main() {
 
     testWidgets(
       'can navigate forwards',
+      // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/inspector/inspector_error_navigator_test.dart:71]:
```suggestion
      // Intentionally unwaited.
```

File: packages/devtools_app/test/inspector/inspector_error_navigator_test.dart [hunk 3]
```diff
@@ -79,6 +81,8 @@ void main() {
 
     testWidgets(
       'can navigate backwards',
+      // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/inspector/inspector_error_navigator_test.dart:84]:
```suggestion
      // Intentionally unwaited.
```

File: packages/devtools_app/test/inspector/inspector_error_navigator_test.dart [hunk 4]
```diff
@@ -90,6 +94,8 @@ void main() {
 
     testWidgets(
       'wraps forwards',
+      // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/inspector/inspector_error_navigator_test.dart:97]:
```suggestion
      // Intentionally unwaited.
```

File: packages/devtools_app/test/inspector/inspector_error_navigator_test.dart [hunk 5]
```diff
@@ -101,6 +107,8 @@ void main() {
 
     testWidgets(
       'wraps backwards',
+      // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/inspector/inspector_error_navigator_test.dart:110]:
```suggestion
      // Intentionally unwaited.
```

File: packages/devtools_app/test/inspector/inspector_error_navigator_test.dart (hunk 3 above)
Comment (CoderDake) [reply packages/devtools_app/test/inspector/inspector_error_navigator_test.dart:84]: I think you could also just make the closure async
`(WidgetTester tester) async => testNavigate(`

File: packages/devtools_app/test/provider/provider_screen_test.dart [hunk 6]
```diff
@@ -27,7 +27,7 @@ void main() {
   late Widget providerScreen;
   late BannerMessagesController bannerMessagesController;
 
-  setUpAll(() => loadFonts());
+  setUpAll(() async => await loadFonts());
```
Comment (CoderDake) [inline packages/devtools_app/test/provider/provider_screen_test.dart:30]:
```suggestion
  setUpAll(() async => loadFonts());
```

File: packages/devtools_app/test/shared/instance_viewer_test.dart [hunk 7]
```diff
@@ -121,7 +121,7 @@ final enumValueInstance = AsyncValue.data(
 );
 
 void main() {
-  setUpAll(() => loadFonts());
+  setUpAll(() async => await loadFonts());
```
Comment (CoderDake) [inline packages/devtools_app/test/shared/instance_viewer_test.dart:124]:
```suggestion
  setUpAll(() async => loadFonts());
```

File: packages/devtools_app/test/shared/service_extension_widgets_test.dart [hunk 8]
```diff
@@ -30,6 +30,9 @@ void main() {
 
     setUp(() {
       reloads = 0;
+
+      // Intebtionally unwaited.
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/shared/service_extension_widgets_test.dart:34]: typo here and elsewhere (in this file and others)

File: packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart [hunk 9]
```diff
@@ -54,12 +56,16 @@ void main() {
...
+    // Intebtionally unwaited.
+    // ignore: discarded_futures
     when(mockClassObject.requestRetainingPath()).thenAnswer((_) async {
       retainingPathNotifier.value = testRetainingPath;
     });
 
     when(mockClassObject.inboundReferences).thenReturn(inboundRefsNotifier);
 
+    // Intebtionally unwaited.
+    // ignore: discarded_futures
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart:68]: is there a way we can turn this lint off for the test/ directory?

File: packages/devtools_app/test/shared/service_extension_widgets_test.dart (hunk 8 above)
Comment (CoderDake) [inline packages/devtools_app/test/shared/service_extension_widgets_test.dart:34]:
```suggestion
      // Intentionally unwaited.
```
y

Comment (CoderDake) [reply packages/devtools_app/test/shared/service_extension_widgets_test.dart:34]: You could also make the setup closure async instead
`    setUp(() async {`

File: packages/devtools_app/test/shared/service_extension_widgets_test.dart [hunk 10]
```diff
@@ -81,6 +84,9 @@ void main() {
 
     setUp(() {
       restarts = 0;
+
+      // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/shared/service_extension_widgets_test.dart:88]:
```suggestion
      // Intentionally unwaited.
```

Comment (CoderDake) [reply packages/devtools_app/test/shared/service_extension_widgets_test.dart:88]: same about the setup could instead be async

File: packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart [hunk 11]
```diff
@@ -38,6 +38,8 @@ void main() {
     when(mockClassObject.reachableSize).thenReturn(null);
     when(mockClassObject.retainedSize).thenReturn(null);
 
+    // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart:41]:
```suggestion
    // Intentionally unwaited.
```

Comment (CoderDake) [reply packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart:41]: you could make setup async instead
`  setUp(() async {`

File: packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart [hunk 12]
```diff
@@ -54,12 +56,16 @@ void main() {
...
+    // ignore: discarded_futures
     when(mockClassObject.requestRetainingPath()).thenAnswer((_) async {
       retainingPathNotifier.value = testRetainingPath;
     });
 
     when(mockClassObject.inboundReferences).thenReturn(inboundRefsNotifier);
 
+    // Intebtionally unwaited.
```
Comment (CoderDake) [inline packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart:67]:
```suggestion
    // Intentionally unwaited.
```

**Review by CoderDake — APPROVED** [review]

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**